pkg/odo/cli/list/services/services.go

```diff
@@ -0,0 +1,198 @@
+package services
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/redhat-developer/odo/pkg/api"
+	"github.com/redhat-developer/odo/pkg/log"
+	"github.com/redhat-developer/odo/pkg/machineoutput"
+	"github.com/redhat-developer/odo/pkg/odo/cli/ui"
+	"github.com/redhat-developer/odo/pkg/odo/cmdline"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions/clientset"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	ktemplates "k8s.io/kubectl/pkg/util/templates"
+	"os"
+	"reflect"
+	"strings"
+)
+
+const RecommendedCommandName = "services"
+
+var (
+	listExample = ktemplates.Examples(`
+	# List the bindable Operator backed services from current namespace
+    %[1]s
+
+	# List all the bindable Operator backed services from all the namespaces
+	%[1]s --all-namespaces
+	%[1]s -A
+
+	# List the bindable Operator backed services in JSON format
+	%[1]s -o json
+	%[1]s --all-namespaces -o json
+	%[1]s -A -o json`)
+
+	listLongDesc = ktemplates.LongDesc(`
+	List the bindable Operator backed services that could bound to the odo component 
+`)
+)
+
+type ServiceListOptions struct {
+	// clientset
+	clientset *clientset.Clientset
+	// context
+	*genericclioptions.Context
+	// working directory
+	contextDir string
+	// flags
+	namespaceFlag     string
+	allNamespacesFlag bool
+}
+
+var _ genericclioptions.Runnable = (*ServiceListOptions)(nil)
+var _ genericclioptions.JsonOutputter = (*ServiceListOptions)(nil)
+
+func (o *ServiceListOptions) SetClientset(clientset *clientset.Clientset) {
+	o.clientset = clientset
+}
+
+func (o *ServiceListOptions) Complete(cmdline cmdline.Cmdline, _ []string) error {
+	var err error
+	o.contextDir, err = os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	o.Context, err = genericclioptions.New(genericclioptions.NewCreateParameters(cmdline))
+	if err != nil {
+		return err
+	}
+
+	if o.namespaceFlag == "" && !o.allNamespacesFlag {
+		o.namespaceFlag = o.GetProject()
+	}
+
+	return err
+}
+
+func (o *ServiceListOptions) Validate() error {
+	if o.allNamespacesFlag && o.namespaceFlag != "" {
+		return errors.New("cannot use --all-namespaces and --namespace flags together")
+	}
+	return nil
+}
+
+func (o *ServiceListOptions) Run(_ context.Context) error {
+	var s *log.Status
+	if o.allNamespacesFlag {
+		s = log.Spinner("Listing bindable services from all namespaces")
+		defer s.End(false)
+	} else {
+		s = log.Spinner(fmt.Sprintf("Listing bindable services from namespace %q", o.namespaceFlag))
+		defer s.End(false)
+	}
+	services, err := o.run()
+	if err != nil {
+		return err
+	}
+	s.End(true)
+	HumanReadable(services)
+	return nil
+}
+
+func (o *ServiceListOptions) run() ([]unstructured.Unstructured, error) {
+	services := []unstructured.Unstructured{}
+
+	if o.allNamespacesFlag {
+		projects, err := o.clientset.ProjectClient.List()
+		if err != nil {
+			return services, err
```

Comment:
Why returning services and not nil as usual? It would be more readable

Reply:
Makes sense. But now it's different, still not `nil`. PTAL.